### PR TITLE
[docs] fix docs-linkcheckbroken task

### LINKED ISSUE #1421: [docs] make linkcheckbroken fails
See https://github.com/plone/documentation/issues/1212

### PR DESCRIPTION
**Pull request by wolbernd**
Fixes #1421

### CODE REVIEW

**Comment by wolbernd** [comment]
@stevepiercy the task `docs-linkcheckbroken` doesn't seem to be called in github workflows.

File: news/1421.internal [hunk 1]
```diff
@@ -0,0 +1 @@
+fixed broken make task linkcheckbroken
```
Comment (stevepiercy) [inline news/1421.internal:1]:
```suggestion
fixed broken make task docs-linkcheckbroken
```

**Comment by stevepiercy** [comment]
Sorry for the delay in responding.

Let's replace the commented out `test sphinx warnings` job with `make docs-linkcheckbroken`. I would like to catch broken links upstream of the main docs.

https://github.com/plone/plone.restapi/blob/781a7675e6c88de11d5b3c6f48ba10bf04ff46cd/.github/workflows/tests.yml#L73-L75

...would become:

```
      # sphinx linkcheck
      - name: sphinx-linkcheck
        run: if [ "${{ matrix.plone-version }}" == "5.2" ] && [ ${{ matrix.python-version }} == '3.7' ]; then make docs-linkcheckbroken; fi
```

And while we're at it, I think we should bump up the matrix variables to 6.0 and 3.9 in both sphinx tasks, and delete the commented out old command.

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Quick fix here, and see comment in issue.

**Comment by wolbernd** [comment]
@stevepiercy Added the linkchecker to github workflows

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -70,6 +70,6 @@ jobs:
           PYTHON_VERSION: ${{ matrix.python-version }}
           PLONE_VERSION: ${{ matrix.plone-version }}
 
-      # test sphinx warnings
-#      - name: sphinx
-#        run: if [ "${{ matrix.plone-version }}" == "5.2" ] && [ ${{ matrix.python-version }} == '3.7' ]; then ./test-no-sphinx-warnings; fi
+      # test for broken links
+      - name: linkcheck
+        run: if [ "${{ matrix.plone-version }}" == "6.0" ] && [ ${{ matrix.python-version }} == '3.9' ]; then make docs-linkcheckbroken; fi
```
Comment (stevepiercy) [inline .github/workflows/tests.yml:75]: Although it is slightly out of scope, let's clean up the other sphinx task by removing the comment and bumping plone and python versions to align with the latest:

https://github.com/plone/plone.restapi/pull/1422/files/dfb88d317460c824fa9a99abbcc9320e2e40bc2b..553a4bbde7303296586ca7b67d8c4f4cfa538836#diff-1db27d93186e46d3b441ece35801b244db8ee144ff1405ca27a163bfe878957fR59-R60

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Review by stevepiercy — APPROVED** [review]
Thank you!

**Comment by stevepiercy** [comment]
@jenkins-plone-org please run jobs

**Pull request merged**